source/shared/cpp/ObjectModel/HostConfig.cpp

```diff
@@ -565,6 +577,36 @@ unsigned int HostConfig::GetFontWeight(FontStyle style, TextWeight weight) const
     return result;
 }
 
+ContainerStyleDefinition HostConfig::GetContainerStyle(ContainerStyle style) const
```

Comment:
Could you please make changes so that it wouldn't return a big struct?

Reply:
So, the problem with returning a pointer in the SharedModel is that all of the other similar functions in HostConfig (i.e.: `GetImageSizes()`, `GetImageBaseUrl()`, etc.) return the struct too. @paulcam206 was saying that passing by reference starts to cause issues with the lifetime of the pointer. I'm hoping `GetBackgroundColor(style)` and `GetForegroundColors(style)` help with that.  If not, suggestions with how to tackle this?

Reply:
one suggestion is leaving this function as it is, keep the copy of the returned value with ACOHostConfig, and move GetBackGroundColor and GetForegroundColor to platform codes. 
in case of iOS, GetBackgroundColor and GetForegroundColor could directly access the copied value. in that way we keep the copying as one time cost. 
And we can also remove ColorsConfig param from getTextBlockColor after making it an instance function.

---
In reply to: [250766675](https://github.com/Microsoft/AdaptiveCards/pull/2289#discussion_r250766675) [](ancestors = 250766675)